arch/arm/core/aarch32/thread.c

```diff
@@ -101,8 +104,9 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	thread->callee_saved.psp = (uint32_t)iframe;
 	thread->arch.basepri = 0;
 
-#if defined(CONFIG_USERSPACE) || defined(CONFIG_FPU_SHARING)
+#if defined(CONFIG_ARM_STORE_EXC_RETURN)
 	thread->arch.mode = 0;
+	thread->arch.exc_return = CONFIG_ARM_DEFAULT_THREAD_EXC_RETURN;
```

Comment:
I really wonder if we could use CMSIS defines  here, instead of defining the value ourselves.